Common/src/interface_interpolation/CInterpolator.cpp

```diff
@@ -143,253 +130,249 @@ unsigned long CInterpolator::Collect_ElementInfo(int markDonor, unsigned short n
+    const auto iPoint = geom->vertex[val_marker][iVertex]->GetNode();
     if (geom->nodes->GetDomain(iPoint)) {
-      Buffer_Send_GlobalPoint[nLocalVertex] = geom->nodes->GetGlobalIndex(iPoint);
-
-      for (iDim = 0; iDim < nDim; iDim++)
-        Buffer_Send_Coord[nLocalVertex*nDim+iDim] = geom->nodes->GetCoord(iPoint, iDim);
-
-      nNodes = 0;
-      nEdges = geom->nodes->GetnPoint(iPoint);
-
-      for (jEdge = 0; jEdge < nEdges; jEdge++){
-        EdgeIndex = geom->nodes->GetEdge(iPoint, jEdge);
-
-        if( iPoint == geom->edges->GetNode(EdgeIndex,0) )
-          dPoint = geom->edges->GetNode(EdgeIndex,1);
-        else
-          dPoint = geom->edges->GetNode(EdgeIndex,0);
-
-        if ( geom->nodes->GetVertex(dPoint, val_marker) != -1 )
-          nNodes++;
-      }
-
-      Buffer_Send_StartLinkedNodes[nLocalVertex] = nLocalLinkedNodes;
-      Buffer_Send_nLinkedNodes[nLocalVertex]     = nNodes;
-
-      nLocalLinkedNodes += nNodes;
+      iVertex_to_iLocalVertex[iVertex] = nLocalVertex;
+      nLocalVertex++;
+    } else {
+      iVertex_to_iLocalVertex[iVertex] = numeric_limits<unsigned long>::max();
+    }
+  }
 
-      Aux_Send_Map[nLocalVertex] = new unsigned long[ nNodes ];
-      nNodes = 0;
+  // coordinates of all domain vertices on the marker
+  su2activematrix Buffer_Send_Coord(nLocalVertex, nDim);
+  // global point IDs of all domain vertices on the marker
+  su2vector<unsigned long> Buffer_Send_GlobalPoint(nVertex);
 
-      for (jEdge = 0; jEdge < nEdges; jEdge++){
-        EdgeIndex = geom->nodes->GetEdge(iPoint, jEdge);
+  /*--- Assign to each domain vertex on the marker, identified by local point ID,
+   * a set of surface-neighbor vertices on the marker, identified by global point ID. ---*/
+  unordered_map<unsigned long, set<unsigned long> > neighbors;
 
-        if( iPoint == geom->edges->GetNode(EdgeIndex,0) )
-          dPoint = geom->edges->GetNode(EdgeIndex,1);
-        else
-          dPoint = geom->edges->GetNode(EdgeIndex,0);
+  /*--- Define or initialize them. ---*/
+  for (unsigned long iVertex = 0; iVertex < nVertex; iVertex++) {
+    const auto iPoint = geom->vertex[val_marker][iVertex]->GetNode();
+    if (geom->nodes->GetDomain(iPoint)) {
+      unsigned long iLocalVertex = iVertex_to_iLocalVertex[iVertex];
+      Buffer_Send_GlobalPoint[iLocalVertex] = geom->nodes->GetGlobalIndex(iPoint);
+      for (unsigned long iDim = 0; iDim < nDim; iDim++) Buffer_Send_Coord[iLocalVertex][iDim] = geom->nodes->GetCoord(iPoint, iDim);
+      neighbors.insert(pair<unsigned long, set<unsigned long> >(iPoint, set<unsigned long>()));
+    }
+  }
 
-        if ( geom->nodes->GetVertex(dPoint, val_marker) != -1 ){
-          Aux_Send_Map[nLocalVertex][nNodes] = geom->nodes->GetGlobalIndex(dPoint);
-          nNodes++;
+  /*--- Define the neighbors map. ---*/
+  for (unsigned long iElem = 0; iElem < nElems; iElem++) {
+    CPrimalGrid* elem = geom->bound[val_marker][iElem];
```

Comment:
This cannot be made `const CPrimalGrid* elem` at the moment because `CPrimalGrid::GetNode` etc are not marked as `const`.

Reply:
I guess/assume you could make the various `GetNode` functions const :) ... and then `const auto elem` ... but this goes a little beyond PR scope so up to you

Reply:
#1352